GraphAlgorithms/MCSNaiveFinder.cs

```diff
@@ -0,0 +1,111 @@
+﻿namespace GraphAlgorithms
+{
+    public class MCSNaiveFinder
+    {
+        /// <summary>
+        /// Finds a maximum common subgraph of two graphs
+        /// </summary>
+        /// <param name="graph1"></param>
+        /// <param name="graph2"></param>
+        /// <returns>A pair of arrays which contain the vertices in the MCS in both graphs</returns>
+        public static (int[], int[]) Find(Graph graph1, Graph graph2)
+        {
+            int[] maxCommonSubgraph = Array.Empty<int>();
+            int[] verticesInGraph1 = Array.Empty<int>();
+            int[] verticesInGraph2 = Array.Empty<int>();
+
+            var subgraphs1 = GenerateSubgraphs(graph1);
+            var subgraphs2 = GenerateSubgraphs(graph2);
+
+            foreach (var subgraph1 in subgraphs1)
+            {
+                foreach (var subgraph2 in subgraphs2)
+                {
+                    if (AreSubgraphsIsomorphic(subgraph1, graph1, subgraph2, graph2))
+                    {
+                        if (maxCommonSubgraph.Length < subgraph1.Length)
+                        {
+                            maxCommonSubgraph = subgraph1;
+                            verticesInGraph1 = subgraph1;
+                            verticesInGraph2 = subgraph2;
+                        }
+                    }
+                }
+            }
+
+            return (verticesInGraph1, verticesInGraph2);
+        }
+
+        private static List<int[]> GenerateSubgraphs(Graph graph)
+        {
+            int n = graph.Size;
+            var subgraphs = new List<int[]>();
+            for (int r = 1; r <= n; r++)
+            {
+                foreach (var subgraphIndices in GetKCombs(Enumerable.Range(0, n), r))
+                {
+                    subgraphs.Add(subgraphIndices);
+                }
+            }
+
+            return subgraphs;
+        }
+
+        private static bool AreSubgraphsIsomorphic(int[] subgraph1, Graph graph1, int[] subgraph2, Graph graph2)
+        {
+            if (subgraph1.Length != subgraph2.Length)
+                return false;
+
+            int n = subgraph1.Length;
+            var vertices = Enumerable.Range(0, n).ToList();
+
+            foreach (var perm in Permutations(vertices, vertices.Count))
+            {
+                bool isomorphic = true;
+
+                for (int i = 0; i < n; i++)
+                {
+                    var permutedI = perm[i];
+                    for (int j = 0; j < n; j++)
+                    {
+                        var permutedJ = perm[j];
+                        if (graph1.AdjacencyMatrix[subgraph1[i], subgraph1[j]]
```

Comment:
Maybe use areNeightbors method. If we later decide to hcange how neighbors work that would make it easier to implement the changes

Reply:
I think here we have to test if they are "precisely equal", because of the definition of graph isomorphism (or at least a reasonable extension) https://math.stackexchange.com/questions/3201645/expanding-definition-of-simple-graph-isomorphism-to-include-multigraphs